transport/http/inbound.go

```diff
@@ -20,7 +20,72 @@
 
 package http
 
-// Inbound TODO
-type Inbound struct {
-	// TODO
+import (
+	"io"
+	"net"
+	"net/http"
+
+	"golang.org/x/net/context"
+
+	"github.com/yarpc/yarpc-go/transport"
+)
+
+// Inbound builds a new HTTP inbound that listens on the given address.
+func Inbound(addr string) transport.Inbound {
+	return &httpInbound{addr: addr}
+}
+
+type httpInbound struct {
+	addr     string
+	listener net.Listener
+}
+
+func (i *httpInbound) Serve(h transport.Handler) error {
+	var err error
+	i.listener, err = net.Listen("tcp", i.addr)
+	if err != nil {
+		return err
+	}
+
+	server := &http.Server{Addr: i.addr, Handler: httpHandler{h}}
+	return server.Serve(i.listener)
+	// TODO Handle connection close errors
+}
+
+func (i *httpInbound) Close() error {
+	return i.listener.Close()
+}
+
+// httpHandler adapts a transport.Handler into a handler for net/http.
+type httpHandler struct {
+	Handler transport.Handler
+}
+
+func (h httpHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	// TODO Procedure name
+
+	defer req.Body.Close()
```

Comment:
Actually this is the server side:

        // For server requests the Request Body is always non-nil
        // but will return EOF immediately when no body is present.
        // The Server will close the request body. The ServeHTTP
        // Handler does not need to.